bin/MadGraph5_aMCatNLO/cards/production/13TeV/ST_4f_w_lo/ST_4f_w_lo_setscales.f

```diff
@@ -0,0 +1,204 @@
+c
+c     INCLUDE and COMMON
+c
+      include 'genps.inc'
+      include 'nexternal.inc'
+      include 'coupl.inc'
+
+      integer i
+      include 'maxamps.inc'
+      integer idup(nexternal,maxproc,maxsproc)
+      integer mothup(2,nexternal)
+      integer icolup(2,nexternal,maxflow,maxsproc)
+      include 'leshouche.inc'
+      include 'run.inc'
+
+      double precision pmass(nexternal)
+      common/to_mass/  pmass
+
+      real*8 xptj,xptb,xpta,xptl,xmtc
+      real*8 xetamin,xqcut,deltaeta
+      common /to_specxpt/xptj,xptb,xpta,xptl,xmtc,xetamin,xqcut,deltaeta
+
+c
+c     ARGUMENTS
+c      
+      REAL*8 P(0:3,nexternal)
+      REAL*8 rscale
+c
+c     EXTERNAL
+c
+      REAL*8 R2,DOT,ET,ETA,DJ,SumDot,PT
+
+c----------
+c     start
+c----------
+
+      if (dynamical_scale_choice.eq.-1) then
+c         Cluster external states until reducing the system to a 2->2 topology whose transverse mass is used for setting the scale.
+c         This is not done in this file due to the clustering.
+         rscale=0d0
+      elseif(dynamical_scale_choice.eq.1) then
+c         Total transverse energy of the event.         
+          rscale=0d0
+          do i=3,nexternal
+             rscale=rscale+et(P(0,i))
+          enddo      
+      elseif(dynamical_scale_choice.eq.2) then
+c         sum of the transverse mass
+c         m^2+pt^2=p(0)^2-p(3)^2=(p(0)+p(3))*(p(0)-p(3))
+          rscale=0d0
+          do i=3,nexternal
+            rscale=rscale+dsqrt(max(0d0,(P(0,i)+P(3,i))*(P(0,i)-P(3,i))))
+          enddo
+          rscale=rscale
+      elseif(dynamical_scale_choice.eq.3) then
+c         sum of the transverse mass divide by 2
+c         m^2+pt^2=p(0)^2-p(3)^2=(p(0)+p(3))*(p(0)-p(3))
+          rscale=0d0
+          do i=3,nexternal
+            rscale=rscale+dsqrt(max(0d0,(P(0,i)+P(3,i))*(P(0,i)-P(3,i))))
+          enddo
+          rscale=rscale/2d0
+      elseif(dynamical_scale_choice.eq.4) then
+c         \sqrt(s), partonic energy
+          rscale=dsqrt(max(0d0,2d0*dot(P(0,1),P(0,2))))
+      elseif(dynamical_scale_choice.eq.5) then
+c         \decaying particle mass, for decays
+          rscale=dsqrt(max(0d0,dot(P(0,1),P(0,1))))
+      elseif(dynamical_scale_choice.eq.0) then
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+cc      USER-DEFINED SCALE: ENTER YOUR CODE HERE                                 cc
+cc      to use this code you must set                                            cc
+cc                 dynamical_scale_choice = 0                                    cc
+cc      in the run_card (run_card.dat)                                           cc
+ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+c        transverse mass of spectator b quark (keep it last in the proc card)
+c        m^2+pt^2=p(0)^2-p(3)^2=(p(0)+p(3))*(p(0)-p(3))
+         i=nexternal-1
+         rscale=(P(0,i)+P(3,i))*(P(0,i)-P(3,i))
+         rscale=4d0*dsqrt(max(0d0,rscale))         

```

Comment:
Not sure I understand the implementation. Is that supposed to be eq 3.1 of the paper?

Reply:
Yes, it is supposed to be eq3.1, I followed other examples from the LO card where `m^2+pt^2=p(0)^2-p(3)^2=(p(0)+p(3))*(p(0)-p(3))`, where the mass and the pT are of the  spectator b-quark